### ISSUE TITLE
Controllers not emitting redstone signal

### ISSUE THREAD

**Issue by ShadowDrakken** [Thutmose/Thut#86]
Mod Version: 6.0.10

I'm having trouble with controllers not emitting redstone signals when the elevator is on their floor. This seems to happen if ANY controller linked to the elevator is setup as a call button. All of the controllers stop emitting and once they stop they can't be coaxed into working again short of breaking and remaking the elevator.

**Comment by ShadowDrakken** [comment]
Update, seems to have nothing to do with being call buttons or not. It just randomly starts acting up while setting up the elevator controls. What's worse is that while setting things up it keeps trying to switch to setting things to call buttons even though it's linked to an elevator... so like one click sets the floor, the next click on another controller sets it to a call button (but says the floor is set) the next click outputs a message that it's changing the call button status but doesn't change the display... it's just weird and random.

This is on the FTB Horizons 3 pack btw.

**Comment by Thutmose** [comment]
I will look into this, last time I checked the redstone signals were working, but I don't remember if I have used an iron door elevator since 6.0.10

**Comment by Thutmose** [comment]
this is working on my end: 

![image](https://user-images.githubusercontent.com/2363472/34450319-0d69aad4-ecd4-11e7-837d-fa91b0c7e2ba.png)

Can you supply screenshots of the setup you have?

**Comment by ShadowDrakken** [comment]
Reloaded my world and the redstone came on for the top floor, but won't change state... screenshot attached
![2017-12-29_20 13 51](https://user-images.githubusercontent.com/638689/34450664-148089d6-ecd5-11e7-8051-6ae581b4cf55.png)

**Comment by Thutmose** [comment]
odd, are you sure the top two blocks are assigned to the correct floors?

**Comment by ShadowDrakken** [comment]
Yup, they're both assigned as floor 2. Building it was a bit of a mess because the remote was acting buggy the whole time :\

**Comment by Thutmose** [comment]
in what way was it acting buggy?

**Comment by ShadowDrakken** [comment]
The behavior is inconsistent.

This time the remote worked fine while I was recording, although the bug is still happening, so I'm uploading the recording for you (it'll be a few minutes).

Anyway, the first time I built it, the remote kept switching between setting floors and setting call button even though I wasn't clicking on anything else, as described in my earlier post.

**Comment by Thutmose** [comment]
odd, it should only set floors if it is linked to the elevator, and set call buttons if it has been cleared after being linked to the elevator.  It is cleared by shift right clicking it on a block other than a controller

**Comment by ShadowDrakken** [comment]
https://youtu.be/B9gs2fGe-hI

**Comment by Thutmose** [comment]
hmm, I wonder if it is from the order that you made them, I made the elevator, then set the floors/call, I will check the order you used

**Comment by Thutmose** [comment]
Hmm, I made in the same order that you did, yet it still seems to work fine on my end... I wonder if another mod is interfering? do things work after relogging?

I also noticed that the numbers on your elevator controllers didn't update with the shading light they should have.

**Comment by ShadowDrakken** [comment]
I generally make the elevator first as well, it doesn't seem to make a difference. If it's a mod interaction bug, then just installing FTB Horizons 3 will give you the same setup I have going. Relogging will update the redstone state, but it won't respond to changes again until relogging.

The weird thing is I have a similar elevator on my server, same mod pack, that's working fine.

One thing I did differently was that originally I made this one with a controller built in, but since that controller never functioned I took it apart and redesigned it. That's when I noticed the issue start, but I can't confirm that as the cause. And it seems that if it is the cause, it's permanently effected my save file

**Comment by ShadowDrakken** [comment]
Ok, so after much testing I'm now able to consistently break it, but not permanently as with my earlier issue. I created a new world to test if the world was damaged, and sure enough this fixed all the apparently permanent issues.

Then to recreate the issue I built several elevators in various orders, and even made sure if it was maybe the chisels & bits I was using that broke things, but to no avail... it just wouldn't break again. That is, until I accidentally set a controller to Floor 1 at a different level than my actual Floor 1. At that point, every controller in that vertical column was broken until I broke and replaced the block. It didn't effect other controllers on any floors though as long as they were in a different column.

Basically, the column had Floor 1 at two different heights, and even remembered the original Floor 1, which causes none of them in that column to understand where the elevator actually was.

I think this is probably related to the earlier issue, but I must have messed up multiple columns and confused the entire system. I think some simple checks in the code to ensure that two controllers at different heights aren't trying to set the same floor number should resolve the issue. Either fail to create the new floor (after making sure the original still exists that is) or delete the original if a new one with the same number is made.

**Comment by Thutmose** [comment]
hmm, I have made elevators where the floor assignements were different y values, this was intended behavior to allow for things like this:

https://youtu.be/Y1G1D36Ltqo

I will look into bugs in these cases.

**Comment by ShadowDrakken** [comment]
No, I mean suppose you have only one controller per floor, and you accidentally assign both heights to the same number. That's where things started acting wonky and stopped being fixable.

**Comment by Thutmose** [comment]
the video I linked had 4 controllers, each pair at the same y level, and set to a different floor, the top two are set to floors 1 and 2, and the bottom two are also assigned to floors 1 and 2, and it works out fine, the last controller set the the floor is where the y level of the floor is set based on

Edit: Do you mean first assign both floors to say 1, then reassign one to 2?  That is supposed to work fine, I can look into testing that when I get time later today, I suspect there may be some cases where that could mess things up for that elevator

**Comment by Thutmose** [comment]
so I downloaded FTB Horizons 3, then made the following:

![image](https://user-images.githubusercontent.com/2363472/34456174-1f16aebe-ed5e-11e7-9ecf-45f4848e7ec6.png)

I linked both floors as 1, then relinked the bottom as 2, and it worked fine, can you try making the exact same setup and report what happens on your end?
